### PR TITLE
INTMDB-741: delete cdk folder

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ INTMDB-741

This PR deletes the CDK folder now that we have deprecated the old constructors and move to the new repository.
I created the branch [cdk_branch_do_not_delete](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/cdk_branch_do_not_delete) to easily access again the CDK folder if needed in the future.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
I think `dependency-review` is failing because this PR has too many changes. I will merge this PR (once I get the approval) and see if the test keeps failing the next PR.

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by maastha — APPROVED** [review]
Great that we're retaining in a separate branch, LGTM

**Pull request merged**
